ssz/sedes/boolean.py

```diff
@@ -19,16 +19,40 @@ def serialize(self, obj):
         else:
             raise Exception("Invariant: no other options for boolean values")
 
-    def deserialize(self, serialized_obj):
-        if serialized_obj == b'\x00':
-            return False
-        elif serialized_obj == b'\x01':
-            return True
+    def deserialize_segment(self, serialized_obj, start_index):
+        """
+        Deserialize the data from the given start_index
+        """
+        # Make sure we have sufficient data for inferring length of list
+        if len(serialized_obj) < start_index + 1:
+            raise DeserializationError(
+                'Insufficient data for deserializing',
+                serialized_obj
+            )
+
+        # Deal with only the first byte of the whole serialized data
+        boolean_serialized_obj = serialized_obj[start_index: start_index + 1]
```

Comment:
@jannikluhn this won't work because of the following example.
Say the string is `str = \x00\x01`. Now if we want to get the first byte, then
```
str[0] = 0
str[0:1] = \x00
```
And the `boolean sedes` recognizes only `\x00` or `\x01` for deserialization.